core/workspace_svg.js

```diff
@@ -288,24 +288,34 @@ Blockly.WorkspaceSvg.prototype.targetWorkspace = null;
  * @type {SVGMatrix}
  * @private
  */
-Blockly.WorkspaceSvg.prototype.inverseScreenCTM_ = null;
+Blockly.WorkspaceSvg.prototype.inverseScreenCTM_ = Blockly.DIRTY;
 
 /**
  * Getter for the inverted screen CTM.
  * @return {SVGMatrix} The matrix to use in mouseToSvg
  */
 Blockly.WorkspaceSvg.prototype.getInverseScreenCTM = function() {
+
+  // Defer getting the screen CTM until we actually need it, this should
+  // avoid forced reflows from any calls to updateInverseScreenCTM.
+  if (this.inverseScreenCTM_ == Blockly.DIRTY) {
+    var ctm = this.getParentSvg().getScreenCTM();
+    if (ctm) {
+      this.inverseScreenCTM_ = ctm.inverse();
+    } else {
+      // When dirty, and we can't get a CTM, set it to null.
+      this.inverseScreenCTM_ = null;
```

Comment:
ooc, is this a behavior change? Seems like before we just kept the old one, though that was likely to be null in this case anyway?  I'm fine with it this way, just wondering if I should try and keep this in my memory for testing of next release :)

Reply:
Nice catch - it is a minor (unnoticed by me) change, if for some reason the `getParentSvg().getScreenCTM()` is no longer working, we would return `null` instead of the last value (which is probably `null`)

I could "fix" it to not have that behavior change pretty easily by adding a @private boolean `WorkspaceSvg.inverseScreenCTMDirty_`, if you think this edge case is an edge we should be wary of.

Reply:
I vaguely remember times when we've seen it be null... but can't quite remember @rachel-fenichel, do you remember?  I'm thinking maybe when Blockly gets loaded in a hidden iframe.  Does that ring a bell?

Reply:
I also wonder if return the "last time we had a ctm" is a better result than `null` - "no CTM exists right now"

Though it is a change in behavior, which I'm still happy to solve ;)

Reply:
> That being said, our consumer for the screen CTM appears to go do extra work if it's null, so we should keep that in mind.

It's interesting that the method calls the same `getScreenCTM().inverse()` directly without checking for nulls, of course it's using an svg element and not the `WorkspaceSvg`, and it is only happening for mouse events on the svg, so can pretty safely assume the svg is on screen, and the CTM wont be null?

I would feel like in the case where we returned `null` from the workspace, this would also get `null` and throw an error?

Reply:
If it isn't hard (which is sounds like you think it is isn't based on your original suggestion?), I think we'd be best to preserve current behavior :-/